src/Container/InMemoryEventStoreFactory.php

```diff
@@ -154,7 +163,45 @@ public function defaultOptions(): iterable
             'metadata_enrichers' => [],
             'plugins' => [],
             'wrap_action_event_emitter' => true,
+            'transactional' => true,
             'read_only' => false,
         ];
     }
+
+    private function determineEventsForDefaultEmitter(array $config): array
+    {
+        if ($this->isTransactional($config)) {
+            return TransactionalActionEventEmitterEventStore::ALL_EVENTS;
+        }
+
+        return ActionEventEmitterEventStore::ALL_EVENTS;
+    }
+
+    private function createEventStore(array $config): EventStore
+    {
+        if ($this->isTransactional($config)) {
```

Comment:
we should do this outside of this method, so the same method is not called 3 times, when only 1 call is needed